src/internal_events/json_parser.rs

```diff
@@ -1,60 +1,53 @@
+use super::prelude::error_stage;
 use metrics::counter;
 use serde_json::Error;
 use vector_core::internal_event::InternalEvent;
 
 #[derive(Debug)]
-pub struct JsonParserFailedParse<'a> {
+pub struct JsonParserError<'a> {
     pub field: &'a str,
     pub value: &'a str,
     pub error: Error,
     pub drop_invalid: bool,
 }
 
-impl<'a> InternalEvent for JsonParserFailedParse<'a> {
+impl<'a> InternalEvent for JsonParserError<'a> {
     fn emit_logs(&self) {
-        if self.drop_invalid {
-            debug!(
-                message = "Event failed to parse as JSON.",
-                field = %self.field,
-                value = %self.value,
-                error = ?self.error,
-                internal_log_rate_secs = 30
-            )
-        } else {
-            warn!(
-                message = "Event failed to parse as JSON.",
-                field = %self.field,
-                value = %self.value,
-                error = ?self.error,
-                internal_log_rate_secs = 30
-            )
-        }
-    }
-
-    fn emit_metrics(&self) {
-        counter!("processing_errors_total", 1,
-            "error_type" => "failed_parse",
+        error!(
+            message = %format!("Event failed to parse as JSON: {:?}", self.error),
+            field = %self.field,
+            value = %self.value,
+            error = "invalid_json",
+            error_type = "parser_failed",
+            stage = error_stage::PROCESSING,
+            drop_invalid = self.drop_invalid,
+            internal_log_rate_secs = 30,
         );
     }
-}
-
-#[derive(Debug)]
-pub struct JsonParserTargetExists<'a> {
-    pub target_field: &'a str,
-}
-
-impl<'a> InternalEvent for JsonParserTargetExists<'a> {
-    fn emit_logs(&self) {
-        warn!(
-            message = "Target field already exists.",
-            target_field = %self.target_field,
-            internal_log_rate_secs = 30
-        )
-    }
 
     fn emit_metrics(&self) {
-        counter!("processing_errors_total", 1,
-            "error_type" => "target_field_exists",
+        counter!(
+            "component_errors_total", 1,
+            "error" => "invalid_json",
+            "error_type" => "parser_failed",
+            "stage" => error_stage::PROCESSING,
+            "field" => self.field.to_string(),
+            "value" => self.value.to_string(),
```

Comment:
@jdrouet apologies, I didn't note this in my review, but I think we should drop the `value` tag here. Similar to my note about `error` this would have infinite cardinality.

Reply:
From what I can tell, `value` will literally be an input string which will have infinite cardinality. We should tag it on the log event, for sure, but I don't think it makes sense to tag this metric with event values.